src/meta-client/src/client.rs

```diff
@@ -88,9 +99,12 @@ impl MetaClientBuilder {
             MetaClient::new(self.id)
         };
 
-        if let (false, false, false) =
-            (self.enable_heartbeat, self.enable_router, self.enable_store)
-        {
+        if let (false, false, false, false) = (
```

Comment:
Why not `if !(self.enable_heartbeat || self.enable_router || self.enable_store || enable_lock)`? Looks wried.